internal/dcache/buffer.go

```diff
@@ -43,26 +44,68 @@ import (
 
 //go:generate $ASSERT_REMOVER $GOFILE
 
-type bufferPool struct {
+var BufPool *BufferPool
+
+type BufferPool struct {
 	pool       sync.Pool    // sync.Pool to relieve GC
 	bufSize    int          // size of buffers in this pool
 	maxBuffers int64        // max allocated buffers allowed
 	curBuffers atomic.Int64 // buffers currently allocated
 }
 
-func NewBufferPool(bufSize int, maxBuffers int) *bufferPool {
-	return &bufferPool{
+func CreateBufferPool(bufSize uint64) error {
```

Comment:
InitBufferPool 
and let's add some comments above this explaining that this is a singleton pool which can be used by callers to allocate buffers of 'bufSize' using the GetBuffer() and PutBuffer() functions.
Also let's keep BufPool private and make GetBuffer() and PutBuffer() package functions.